src/java.base/share/classes/javax/crypto/EncryptedPrivateKeyInfo.java

```diff
@@ -308,6 +321,253 @@ private PKCS8EncodedKeySpec getKeySpecImpl(Key decryptKey,
+     * @throws NullPointerException if the {@code key} or {@code encKey} are
+     * null. If {@code params} is non-null, {@code algorithm} cannot be
+     * {@code null}.
+     *
+     * @implNote The `jdk.epkcs8.defaultAlgorithm` Security Property defines
+     * the default encryption algorithm and the {@code AlgorithmParameterSpec}
+     * are the provider's algorithm defaults.
+     *
+     * @since 25
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.PEM_API)
+    public static EncryptedPrivateKeyInfo encryptKey(PrivateKey key, Key encKey,
+        String algorithm, AlgorithmParameterSpec params, Provider provider,
+        SecureRandom random) {
+
+        Objects.requireNonNull(key);
+        Objects.requireNonNull(encKey);
+        if (algorithm == null) {
+            if (params != null) {
+                throw new NullPointerException("algorithm must be specified " +
+                    "if params is non-null.");
+            }
+
+            if (Pem.DEFAULT_ALGO == null || Pem.DEFAULT_ALGO.length() == 0) {
+                throw new RuntimeException("Security property " +
+                    "\"jdk.epkcs8.defaultAlgorithm\" may not specify a " +
+                    "valid algorithm.  Operation cannot be performed.");
+            }
+            algorithm = Pem.DEFAULT_ALGO;
+        }
+        return encryptKeyImpl(key, algorithm, encKey, params, provider, random);
+    }
+
+    private static EncryptedPrivateKeyInfo encryptKeyImpl(PrivateKey key,
+        String algorithm, Key encryptKey, AlgorithmParameterSpec params,
+        Provider provider, SecureRandom random) {
+        AlgorithmId algId;
+        byte[] encryptedData;
+        Cipher c;
+        DerOutputStream out;
+
+        if (random == null) {
+            random = JCAUtil.getDefSecureRandom();
+        }
+        try {
+            if (provider == null) {
+                c = Cipher.getInstance(algorithm);
+            } else {
+                c = Cipher.getInstance(algorithm, provider);
+            }
+            c.init(Cipher.ENCRYPT_MODE, encryptKey, params, random);
+            encryptedData = c.doFinal(key.getEncoded());
+            algId = new AlgorithmId(Pem.getPBEID(algorithm), c.getParameters());
+            out = new DerOutputStream();
+            algId.encode(out);
+            out.putOctetString(encryptedData);
+        } catch (InvalidAlgorithmParameterException | NoSuchAlgorithmException |
+                 NoSuchPaddingException e) {
+            throw new IllegalArgumentException(e);
+        } catch (IllegalBlockSizeException | BadPaddingException |
+                 InvalidKeyException e) {
+            throw new RuntimeException(e);
+        }
+        return new EncryptedPrivateKeyInfo(
+            DerValue.wrap(DerValue.tag_Sequence, out).toByteArray(),
+            encryptedData, algId, c.getParameters());
+    }
+
+    /**
+     * Returns a {@code PrivateKey} from the encrypted data in this instance.
```

Comment:
Follow existing `get` methods, "Extract the enclosed PrivateKey object from the encrypted data and return it." Same with the other `getKey` method.